### PR TITLE
Use the flip-to-first-page workaround more widely

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

It needs to be executed for contact creation forms too.

medic/medic-webapp#3928

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

**Comment by garethbowen** [comment]
Merging immediately so we can get another 2.13.0 beta out.

File: static/js/services/enketo.js [hunk 1]
```diff
@@ -343,7 +342,9 @@ angular.module('inboxServices').service('Enketo',
         })
         .then(transformXml)
         .then(function(doc) {
-          var form = renderFromXmls(doc, selector, instanceData);
+          return renderFromXmls(doc, selector, instanceData);
+        })
+        .then(function(form) {
```
Comment (alxndrsn) [inline static/js/services/enketo.js]: For my understanding: this doesn't actually change anything, does it?  (except add more promisey steps)

File: static/js/services/enketo.js [hunk 2]
```diff
@@ -315,8 +315,13 @@ angular.module('inboxServices').service('Enketo',
         })
         .then(function(doc) {
           replaceJavarosaMediaWithLoaders(id, doc.html);
-          var form = renderFromXmls(doc, selector, instanceData);
+          return renderFromXmls(doc, selector, instanceData);
+        })
+        .then(function(form) {
```
Comment (alxndrsn) [inline static/js/services/enketo.js]: Again, this is just introducing another promise step for no specific reason (except for readability?) as far as I can tell.  Is that correct?

Comment (garethbowen) [reply static/js/services/enketo.js]: Actually no. I realised while looking at this code that `renderFromXmls` is actually async so not having this change could actually cause an issue.

File: static/js/services/enketo.js (hunk 1 above)
Comment (garethbowen) [reply static/js/services/enketo.js]: As above.

File: static/js/services/enketo.js (hunk 2 above)
Comment (alxndrsn) [reply static/js/services/enketo.js]: So `form` would have previously been a `Promise`, then?  Looks like the old naming was a bit weird, but now makes sense :slightly_smiling_face:

Comment (garethbowen) [reply static/js/services/enketo.js]: Right - I think it used to be a sync function and obviously wasn't updated/renamed when it was changed to async...

**Pull request merged**